src/main/java/io/vertx/core/net/TCPSSLOptions.java

```diff
@@ -405,6 +410,50 @@ public TCPSSLOptions addCrlPath(String crlPath) throws NullPointerException {
   }
 
   /**
+   * Add key options for SNI host.
+   *
+   * @param domain The domain name to match. May contain wildcards.
+   * @param keyCertOptions The key options to use when matching this host.
+   * @return a reference to this, so the API can be used fluently
+   */
+  public TCPSSLOptions addSNIKeyCertOptionsForDomain(String domain, KeyCertOptions keyCertOptions) throws NullPointerException {
```

Comment:
That's not what I meant.i I'm talking about the type of the return value. What I mean is that there is no way that getSniPemCertKeyOptions can return the type of Map<String, PemCertKeyOptions>, because it is stored as Map <String, KeyCertOptions >. This is why there is only one getter for the non -sni options as well getKeyCertOptions